rotkehlchen/data_import/importers/_import_helpers.py

```diff
@@ -0,0 +1,241 @@
+    raise ValueError(
+        f'Could not classify transaction between {origin} and {destination} '
+        f'with type {transaction_type}',
+    )
+
+
+def check_transaction_type(
+    transaction_type: str,
+    origin: str,
+    destination: str,
+    target_platform: str,
+) -> None:
+    """Check if the transaction type is valid."""
+    if is_on_exchange(origin, destination):
+        return
+    if transaction_type == 'in' and origin != target_platform:
+        raise ValueError(
+            'Transaction Type and origin don\'t match. '
+            f'Transaction Type is {transaction_type} but origin is {origin}',
+        )
+    if transaction_type == "out" and destination != target_platform:
+        raise ValueError(
+            'Transaction Type and destination don\'t match. '
+            f'Transaction Type is {transaction_type} but destination is {destination}',
+        )
+
+
+def is_on_exchange(origin: str, destination: str) -> bool:
+    """Check if the transaction is on exchange."""
+    return origin == destination
+
+
+def is_same_asset(origin_asset: Asset, destination_asset: Asset) -> bool:
+    """Check if the transaction has the same origin and destination assets."""
+    return origin_asset == destination_asset
+
+
+def is_withdrawal(
+    transaction_type: str,
+    origin_asset: Asset,
+    destination_asset: Asset,
+) -> bool:
+    """Check if the transaction is a withdrawal."""
+    if not is_same_asset(origin_asset, destination_asset):
+        raise ValueError(
+            'Transaction Type and origin/destination assets don\'t match. '
+            f'Transaction Type is {transaction_type} but origin asset is {origin_asset} '
+            f'and destination asset is {destination_asset}',
+        )
+    # Add negative check to serve for 'is_deposit'
+    return transaction_type == 'out' and transaction_type != 'in'
+
+
+def is_deposit(
+    transaction_type: str,
+    origin_asset: Asset,
+    destination_asset: Asset,
+) -> bool:
+    """Check if the transaction is a deposit."""
+    return not is_withdrawal(transaction_type, origin_asset, destination_asset)
+
+
+def is_trade(origin_asset: Asset, destination_asset: Asset) -> bool:
+    """Check if the transaction is a trade."""
+    return not is_same_asset(origin_asset, destination_asset)
+
+
+def is_ledger_action(
+    origin_asset: Asset,
+    destination_asset: Asset,
+    origin_amount: AssetAmount,
+    destination_amount: AssetAmount,
+) -> bool:
+    """Check if the transaction is a ledger action."""
+    return origin_asset == destination_asset and origin_amount == destination_amount
+
+
+def get_action_type(transaction_type: str) -> LedgerActionType | None:
+    if transaction_type == 'in':
+        return LedgerActionType.INCOME
+    if transaction_type == 'out':
+        return LedgerActionType.EXPENSE
+    log.debug(f'Ignoring uncaught transaction type of {transaction_type}.')
+    return None
+
+
+class TransactionType:
```

Comment:
Hmmm this structure looks heavy. Why not to use an Enum? We already have one https://github.com/rotki/rotki/blob/d78b05575cd1299967db1bf7005a053778cf50f9/rotkehlchen/accounting/mixins/event.py#L14